### PR TITLE
build compile transformer now rewrites bare module specifiers to paths.

### PR DESCRIPTION
**Pull request by aomarks**
Also:

- Bare specifier rewriter now works on Windows.
- Bare specifier rewriter no longer tries to resolve fully qualified URLs.
- Added `moduleResolution` option to `OptimizeOptions`.
- Combined compile and minimize babel transforms.

 - [ ] CHANGELOG.md has been updated

### CODE REVIEW

**Comment by justinfagnani** [comment]
We should add a new option to match polyserve and the bundler: `moduleResolution: 'node'|'none'`

**Comment by aomarks** [comment]
> We should add a new option to match polyserve and the bundler: moduleResolution: 'node'|'none'

Done PTAL

File: src/babel-plugin-bare-specifiers.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import {dirname, relative} from 'path';
 import resolve = require('resolve');
 import {NodePath} from 'babel-traverse';
+import * as url from 'url';
```
Comment (justinfagnani) [inline src/babel-plugin-bare-specifiers.ts]: use `whatwg-url`

File: src/babel-plugin-bare-specifiers.ts [hunk 2]
```diff
@@ -52,6 +57,10 @@ export const resolveBareSpecifiers =
           let relativeSpecifierUrl =
               relative(dirname(filePath), resolvedSpecifier);
 
+          if (process.platform === 'win32') {
```
Comment (justinfagnani) [inline src/babel-plugin-bare-specifiers.ts]: consider using the is-windows package

File: src/optimize-streams.ts [hunk 3]
```diff
@@ -119,17 +110,25 @@ export class GenericOptimizeTransform extends Transform {
...
 export class JSCompileTransform extends GenericOptimizeTransform {
-  constructor() {
-    const transformer = (content: string, file: File) =>
-        babelTransform(content, {
-          presets: [babelPresetES2015NoModules],
-          plugins: [
+  constructor(options: OptimizeOptions['js']) {
+    const transformer = (content: string, file: File) => {
+      const presets = [];
+      const plugins = [
```
Comment (justinfagnani) [inline src/optimize-streams.ts]: add comment explaining why we're loading sync plugins without the transform. Also, make these a single array to use with the minify transform.

File: src/optimize-streams.ts [hunk 4]
```diff
@@ -143,8 +142,8 @@ export class JSMinifyTransform extends GenericOptimizeTransform {
         babelTransform(content, {
           presets: [minifyPreset(null, {simplifyComparisons: false})],
           plugins: [
-            babelPluginSyntaxObjectRestSpread,
-            babelPluginSyntaxDynamicImport,
+            objectRestSpreadSyntax,
```
Comment (justinfagnani) [inline src/optimize-streams.ts]: Add todo/issue to combine with JSCompileTransform

Comment (aomarks) [reply src/optimize-streams.ts]: Just went ahead and did it.

File: src/optimize-streams.ts (hunk 3 above)
Comment (aomarks) [reply src/optimize-streams.ts]: Done.

File: src/babel-plugin-bare-specifiers.ts (hunk 2 above)
Comment (aomarks) [reply src/babel-plugin-bare-specifiers.ts]: Done.

File: src/babel-plugin-bare-specifiers.ts (hunk 1 above)
Comment (aomarks) [reply src/babel-plugin-bare-specifiers.ts]: Done.

**Comment by aomarks** [comment]
PTAL

**Comment by justinfagnani** [comment]
LGMT still

**Pull request merged**